Modules/posixmodule.c

```diff
@@ -5266,7 +5266,7 @@ os_spawnve_impl(PyObject *module, int mode, path_t *path, PyObject *argv,
             goto fail_1;
         }
         if (i == 0 && !argvlist[0][0]) {
-            lastarg = i;
+            lastarg = i + 1;
```

Comment:
You can restore the initial value to `Py_ssize_t lastarg = 0`. But leaving it as -1 shouldn't be a problem since failures before this loop go to `fail_0` instead of `fail_1`.